packages/frontend-api/src/Model/Product/BatchLoad/ProductElasticsearchBatchProvider.php

```diff
@@ -143,4 +161,25 @@ protected function getFilterQueryForBrand(ProductBatchLoadByEntityData $productB
             $productBatchLoadByEntityData->getEntity($this->entityNameResolver->resolve(Brand::class)),
         );
     }
+
+    /**
+     * @param array $productIds
+     * @param \Shopsys\FrameworkBundle\Model\Category\Category $category
+     * @return \Shopsys\FrameworkBundle\Model\Product\Search\FilterQuery
+     */
+    protected function getSellableByProductsIdsInCategoryFilterQuery(
+        array $productIds,
+        Category $category,
+    ): FilterQuery {
+        $filterQuery = $this->filterQueryFactory->createSellableProductsByProductIdsFilter(
+            $productIds,
+            $this->productFrontendLimitProvider->getProductsFrontendLimit(),
+        );
+
+        foreach ($this->categoryAutomatedFilterFacade->getByCategory($category) as $automatedFilter) {
+            $filterQuery = $automatedFilter->applyFilter($filterQuery);
+        }
```

Comment:
this is a similar code as in `FilterQueryFactory::filterByCategory()`. Isn't possible to unify it? also here I would expect filterByCategory called as the method is named `...InCategory...`

Reply:
nice catch! what about this? d4d0b51c9b

Reply:
and 758c55d3c9

Reply:
👍